.github/workflows/pr_build.yml

```diff
@@ -76,6 +76,41 @@ jobs:
           # upload test reports only for java 17
           upload-test-reports: ${{ matrix.java_version == '17' }}
 
+  linux-test-native-datafusion-scan:
+    env:
+      COMET_PARQUET_SCAN_IMPL: "native_datafusion"
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        java_version: [17]
+        test-target: [rust, java]
+        spark-version: ['3.5']
+        scala-version: ['2.12']
```

Comment:
What about other versions? Spark 3.4, 4.0, Scala 2.13?

Reply:
I was thinking that this would be enough for now to prevent regressions while this is still an experimental feature.

We could increase the matrix here but it will add overhead to all PRs. What do you think @mbutrovich?

Reply:
I have not tested Spark 3.5, Spark 4.0, Scala 2.13 with `native_datafusion` and would prefer not to get spread too thin if we get a deluge of issues. I propose walking before we run and get this pipeline in great shape and then add more.

Reply:
@kazuyukitanimura Could we merge this PR to add these initial checks. We currently have almost zero tests running in CI for `native_datafusion`, so this is a big step forward. It is just the first step though.

Reply:
Sure, would you mind filing an issue to track this? I will go ahead and approve

Reply:
Thanks. I filed https://github.com/apache/datafusion-comet/issues/1486